File: android/app/src/main/java/voidawt/image/DataBufferInt.java
package voidawt.image;

public class DataBufferInt extends DataBuffer {
    private final int[] data;

    public DataBufferInt(int[] data, int size) {
        this.data = data;
        this.size = size;
    }

    public DataBufferInt(int size) {
        this.data = new int[size];
        this.size = size;
    }

    public int[] getData() {
        return data;
    }
}

File: src/commonMain/kotlin/awt/image/DataBufferInt.kt
package awt.image

expect class DataBufferInt {
    constructor(size: Int)
    constructor(dataArray: IntArray, size: Int)

    fun getData(): IntArray
    fun getElem(i: Int): Int
    fun setElem(i: Int, value: Int)
}

expect val DataBufferInt.bufferSize: Int